### PR TITLE
sql: implement `ALTER MATERIALIZED VIEW` and `DROP MATERIALIZED VIEW`

### LINKED ISSUE #52733: sql: implement `ALTER MATERIALIZED VIEW` and `DROP MATERIALIZED VIEW`
These commands are just translated to the standard `ALTER VIEW/DROP VIEW` commands. This can be done once #52530 lands.

### PR DESCRIPTION
**Pull request by rohany**
Fixes #52733.

Release note (sql change): Implement the `ALTER MATERIALIZED VIEW` and
`DROP MATERIALIZED VIEW` commands.

### CODE REVIEW

File: pkg/sql/parser/sql.y [hunk 1]
```diff
@@ -6579,12 +6617,36 @@ alter_rename_view_stmt:
...
+    newName := $7.unresolvedObjectName()
+    $$.val = &tree.RenameTable{
+      Name: name,
+      NewName: newName,
+      IfExists: false,
+      IsView: true,
+      IsMaterialized: true,
+    }
+  }
 | ALTER VIEW IF EXISTS relation_expr RENAME TO view_name
```
Comment (otan) [inline pkg/sql/parser/sql.y:6632]: can we just do
```
| ALTER opt_materialized VIEW IF EXISTS ...
    name := $6.unresolvedObjectName()
    newName := $9.unresolvedObjectName()
    $$.val = &tree.RenameTable{Name: name, NewName: newName, IfExists: true, IsView: true, IsMaterialized: $2.bool()}
```
?

File: pkg/sql/views.go [hunk 2]
```diff
@@ -57,3 +60,23 @@ func (d planDependencies) String() string {
 	}
 	return buf.String()
 }
+
+// checkViewMaterialized ensures that if a view is required, then the view
+// is materialized or not as desired.
+func checkViewMaterialized(desc sqlbase.TableDescriptor, requireView, wantMaterialized bool) error {
```
Comment (otan) [inline pkg/sql/views.go:66]: maybe `checkViewMatchesMaterialized`?

**Review by otan — APPROVED** [review]
LGTM with minor comments

File: pkg/sql/parser/sql.y (hunk 1 above)
Comment (rohany) [reply pkg/sql/parser/sql.y:6632]: lol i tried this, but got some nice shift/reduce conflicts so I gave up.

File: pkg/sql/views.go (hunk 2 above)
Comment (rohany) [reply pkg/sql/views.go:66]: that sounds clearer, I'll update

File: pkg/sql/parser/sql.y (hunk 1 above)
Comment (otan) [reply pkg/sql/parser/sql.y:6632]: hmmMmMmMMMMmm

**Pull request merged**